### PR TITLE
#123 --watch-idle flag implementation

### PR DESCRIPTION
**Pull request by daubaris**
Added the flag which allows to ignore incoming processes while some process is still in execution. Not sure if the implementation is correct regarding the 4 scenarios though. Would be great if someone could elaborate on that matter.

### CODE REVIEW

**Comment by passcod** [comment]
Mmm I think a better flag name would be `--watch-when-idle`, which makes it clear what's happening.

File: src/run.rs [hunk 1]
```diff
@@ -240,8 +268,14 @@ impl Handler for ExecHandler {
...
-                signal_process(&self.child_process, None, true);
-                self.spawn(ops)?;
+                if self.args.watch_when_idle {
+                    if !running_process {
+                        self.spawn(ops)?;
+                    }
+                } else {
+                    signal_process(&self.child_process, None, true);
+                    self.spawn(ops)?;
+                }
```
Comment (passcod) [inline src/run.rs]: Can you try instead to add a third boolean to the match and see if that works out better? having the same if three times feels a bit weird

So like 

```rust
let scenario = (self.args.restart, self.signal.is_some(), self.args.watch_when_idle);
match scenario {
    // SIGHUP scenario (needs to be up here to take precedence)
    (false, true, _) => signal_process(&self.child_process, self.signal, false),
    // watch-when-idle
    (_, _, true) => if !running_process { self.spawn(ops)?; },
    // etc
    (true, true, false) => {
        signal_process(&self.child_process, self.signal, true);
        self.spawn(ops)?;
    },
    // ...
}
```

File: src/run.rs [hunk 2]
```diff
@@ -189,6 +189,19 @@ impl ExecHandler {
 
         Ok(())
     }
+
+    pub fn has_running_process(&self) -> bool {
+        let guard = self
+            .child_process
+            .read()
+            .expect("poisoned lock in signal_process");
```
Comment (passcod) [inline src/run.rs]: change that expect message to have the right method name

**Comment by passcod** [comment]
The CI fails so far are unrelated — I'll go fix them elsewhere

File: src/run.rs (hunk 1 above)
Comment (daubaris) [reply src/run.rs]: Yeah, this makes sense..

File: src/run.rs [hunk 3]
```diff
@@ -189,6 +189,19 @@ impl ExecHandler {
...
+        let guard = self
+            .child_process
+            .read()
+            .expect("poisoned lock in has_running_process");
+
+        if let Some(ref _child) = *guard {
+            return true;
+        }
+
+        false
```
Comment (passcod) [inline src/run.rs]: Oh, surely this could be `(*guard).is_some()` or something?

File: src/run.rs [hunk 4]
```diff
@@ -215,31 +228,44 @@ impl Handler for ExecHandler {
         // 3. Just send a specified signal to the child, do nothing more
         // 4. Make sure the previous run was ended, then run the command again
```
Comment (passcod) [inline src/run.rs]: Can you also update this comment given there's now five scenarios

Comment (daubaris) [reply src/run.rs]: Yeah, sorry, my bad..

**Comment by passcod** [comment]
I'll merge this once I've fixed CI

**Pull request merged**